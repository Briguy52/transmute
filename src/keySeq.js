import { Iterable, Seq } from "immutable";

/**
 * Get a Seq of the keys in `value`.
 *
 * @param  {Iterable|Object|Array} value
 * @return {Seq}
 */
export default function keySeq<K>(
  value: Array<*> | Iterable<K, *> | { [key: K]: * }
) {
  if (!Iterable.isIterable(value)) {
    value = Seq(value);
  }
  return value.keySeq();
}
