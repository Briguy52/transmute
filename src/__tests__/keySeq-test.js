import { Map } from "immutable";
import keySeq from "../keySeq";

describe("transmute/keySeq", () => {
  it("gets the keys from an Array", () => {
    expect(keySeq([1, 2, 3])).toMatchSnapshot();
  });

  it("gets the keys from a Map", () => {
    expect(keySeq(Map({ one: 1, two: 2, three: 3 }))).toMatchSnapshot();
  });

  it("gets the keys from an Object", () => {
    expect(keySeq({ one: 1, two: 2, three: 3 })).toMatchSnapshot();
  });
});
