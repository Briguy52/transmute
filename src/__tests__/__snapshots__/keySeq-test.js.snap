// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/keySeq gets the keys from a Map 1`] = `
Array [
  "one",
  "two",
  "three",
]
`;

exports[`transmute/keySeq gets the keys from an Array 1`] = `
Array [
  0,
  1,
  2,
]
`;

exports[`transmute/keySeq gets the keys from an Object 1`] = `
Array [
  "one",
  "two",
  "three",
]
`;
